bodhi/server/scripts/clean_old_mashes.py

```diff
@@ -0,0 +1,67 @@
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+This tool cleans up old mashes that are left over in mash_dir.
+"""
+
+import click
+import os
+import shutil
+
+from bodhi.server import config
+
+
+# How many of the newest mash dirs to keep during cleanup
+NUM_TO_KEEP = 10
+
+
+@click.command()
+@click.version_option(message='%(version)s')
+def clean_up():
+    """
+    Delete any repo mashes that are older than the newest 10 from each repo series.
+    """
+    mash_dir = config.config['mash_dir']
+
+    # This data structure will map the beginning of a group of dirs for the same repo to a list of
+    # the dirs that start off the same way.
+    pattern_matched_dirs = {}
+
+    for directory in [d for d in os.listdir(mash_dir) if os.path.isdir(os.path.join(mash_dir, d))]:
+        # If this directory ends with a float, it is a candidate for potential deletion
+        try:
+            split_dir = directory.split('-')
+            float(split_dir[-1])
+        except ValueError:
+            # This directory didn't end in a float, so let's just move on to the next one.
+            continue
+
+        pattern = directory.replace(split_dir[-1], '')
+        if pattern in pattern_matched_dirs:
+            pattern_matched_dirs[pattern].append(directory)
+        else:
+            pattern_matched_dirs[pattern] = [directory]
+
+    dirs_to_delete = []
+
+    for pattern, dirs in pattern_matched_dirs.items():
```

Comment:
That's a good point. I think I had intended to write ```dirs``` instead of ```pattern_matched_dirs[pattern]``` in the code below when I wrote this line, but then promptly forgot to access the lists that way. Silly me.

Since I really don't need the keys at all (they are really just used to collate the lists of dirs that match the patterns), I've changed this line to:

```
for dirs in pattern_matched_dirs.values():
```

And then I used dirs in the code below. Nice catch on a silly pattern!